### PR TITLE
Feature/bai 673 resuable model release display

### PR DESCRIPTION
**Pull request by ARADDCC002**
Added a release tab to the new model beta page. Each release has its own display.

image

### CODE REVIEW

**Comment by ARADDCC002** [comment]
Now includes a draft new release dialog 

image

File: frontend/src/model/DraftNewReleaseDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import FileInput from 'src/common/FileInput'
+
+export default function DraftNewReleaseDialog({ open, handleClose }: { open: boolean; handleClose: () => void }) {
+  const [releaseName, setReleaseName] = useState('')
+  const [semanticVersion, setSemanticVersion] = useState('')
+  const [releaseNotes, setReleaseNotes] = useState('')
+  const [isMinorRelease, setIsMinorRelease] = useState(false)
+  const [artefact, setArtefact] = useState<any>(undefined)
+
+  function onSubmit(event) {
```
Comment (ARADDCC012) [inline frontend/src/model/DraftNewReleaseDialog.tsx:27]: What type is `event`?

File: frontend/src/model/DraftNewReleaseDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  function handleMinorReleaseChecked() {
+    setIsMinorRelease(!isMinorRelease)
+  }
+
+  function handleArtefactChange(event: ChangeEvent<HTMLInputElement>) {
+    if (event.target.files) setArtefact(event.target.files[0])
+  }
+
+  function validSemver(semverInput: string) {
```
Comment (ARADDCC012) [inline frontend/src/model/DraftNewReleaseDialog.tsx:43]: Rename to `isValidSemver`?

File: frontend/src/model/ModelReleaseDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+import Link from '../Link'
+
+export default function ModelReleaseDisplay({
+  modelUuid,
+  release,
+  latestRelease,
+}: {
+  modelUuid: string
+  release: any
```
Comment (ARADDCC012) [inline frontend/src/model/ModelReleaseDisplay.tsx:15]: Can this be typed?

File: frontend/src/model/ModelReleaseDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    const formattedYear = `'${year.substring(date.getFullYear().toString().length - 2)}`
+    return `${date.getDate()} ${date.toLocaleString('default', { month: 'short' })} ${formattedYear}`
+  }
+
+  function approvalStatus(status: ApprovalStates, label: string) {
+    switch (status) {
+      case ApprovalStates.Accepted:
+        return (
+          <Tooltip title={`${label} has approved this release`}>
+            <Done htmlColor={green[500]} />
```
Comment (ARADDCC012) [inline frontend/src/model/ModelReleaseDisplay.tsx:32]: Could these three icons use theme colours? Success, warning and error are green, orange and red.

File: frontend/src/model/beta/Releases.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import { Box, Button, Stack } from '@mui/material'
+import { useEffect, useState } from 'react'
+
+import DraftNewReleaseDialog from '../DraftNewReleaseDialog'
+import ModelReleaseDisplay from '../ModelReleaseDisplay'
+
+export default function Releases({ model }: { model: any }) {
```
Comment (ARADDCC012) [inline frontend/src/model/beta/Releases.tsx:7]: Can `model` be typed?

File: frontend/src/model/ModelReleaseDisplay.tsx [hunk 6]
```diff
@@ -1,23 +1,23 @@
...
-  release: any
+  modelId: string
+  release: ReleaseInterface
   latestRelease: string
 }) {
   const theme = useTheme()
 
-  function formatDate(timestamp) {
+  function formatDate(timestamp: string) {
+    console.log(timestamp)
```
Comment (ARADDCC012) [inline frontend/src/model/ModelReleaseDisplay.tsx:20]: Oopsy!

File: frontend/src/model/ModelReleaseDisplay.tsx [hunk 7]
```diff
@@ -86,16 +86,16 @@ export default function ModelReleaseDisplay({
                 {latestVersionAdornment()}
                 <Divider orientation='vertical' flexItem />
                 <Stack direction={{ sm: 'row', xs: 'column' }}>
-                  {approvalStatus(release.managerApproved, 'Manager')}
-                  {approvalStatus(release.reviewerApproved, 'Technical Reviewer')}
+                  {approvalStatus(ApprovalStates.Accepted, 'Manager')}
```
Comment (ARADDCC012) [inline frontend/src/model/ModelReleaseDisplay.tsx:89]: Are these states placeholder for now?

**Review by ARADDCC012 — APPROVED** [review]

**Pull request merged**